ably/options.go

```diff
@@ -292,56 +338,87 @@ func (opts *ClientOptions) disconnectedRetryTimeout() time.Duration {
 	return defaultOptions.DisconnectedRetryTimeout
 }
 
-func (opts *ClientOptions) restURL() string {
+func (opts *ClientOptions) getRestHost() string {
 	host := opts.RestHost
-	if host == "" {
-		host = defaultOptions.RestHost
-		if opts.Environment == "production" {
-			opts.Environment = ""
-		}
-		if opts.Environment != "" {
-			host = opts.Environment + "-" + host
-		}
+	defaultHost := defaultOptions.RestHost
+	if empty(host) {
+		host = defaultHost
 	}
-	if opts.NoTLS {
-		port := opts.Port
-		if port == 0 {
-			port = 80
-		}
-		return "http://" + net.JoinHostPort(host, strconv.FormatInt(int64(port), 10))
-	} else {
-		port := opts.TLSPort
-		if port == 0 {
-			port = 443
-		}
-		return "https://" + net.JoinHostPort(host, strconv.FormatInt(int64(port), 10))
+	if host == defaultHost && !opts.isProductionEnvironment() {
+		host = opts.Environment + "-" + host
 	}
+	return host
 }
 
-func (opts *ClientOptions) realtimeURL() string {
+func (opts *ClientOptions) getRealtimeHost() string {
 	host := opts.RealtimeHost
-	if host == "" {
-		host = defaultOptions.RealtimeHost
-		if opts.Environment == "production" {
-			opts.Environment = ""
-		}
-		if opts.Environment != "" {
-			host = opts.Environment + "-" + host
+	defaultHost := defaultOptions.RealtimeHost
+	if empty(host) {
+		if !empty(opts.RestHost) {
+			logger := opts.Logger.Sugar()
+			logger.Warnf(`restHost is set to %s but realtimeHost is not set so setting realtimeHost to %s too. 
+							If this is not what you want, please set realtimeHost explicitly.`, opts.RestHost, opts.RealtimeHost)
```

Comment:
Since this string is enclosed in backticks, whitespace will be preserved and hence will appear in the resulting log message, so I think we should put this all on the line above.

Reply:
Won't it become too long to read? what about concatenation by + sign

Reply:
My screen is quite wide, and my text editor (Vim) is advanced enough to wrap lines if I tell it 🙂

Reply:
Vim 🙏

Reply:
@sacOO7 this has still not been resolved, if you intend to leave it as is please say, but I really think the output will look very odd in a log